beam-migrate/Database/Beam/Migrate/Backend.hs

```diff
@@ -96,7 +97,9 @@ data BeamMigrationBackend be m where
 -- polymorphism
 data SomeBeamMigrationBackend where
   SomeBeamMigrationBackend :: ( BeamMigrateSqlBackend be
-                              , Typeable be )
+                              , Typeable be
+                              , Fail.MonadFail m -- TODO check whether this is good
```

Comment:
Is this needed? It seems odd when there wasn't even a `Monad` constraint before.

Reply:
Hmm now that you mention it I think `MonadFail` could be moved to `BeamMigrationBackend` (scroll up) and `BeamMigrateSqlBackend` can be deleted as its already included in that list. I'm a bit skeptical of all these dictionaries (explicit or implicit), so maybe there is more that could be done but it would a much more invasive.

Taking a step back, the underlying problem is migrations can fail, and `ensureBackendTables`, which is used by default in particular, does call `fail` after the 8.8 fixes. We can use something more structured than `Fail` (always fine by me!) but we need something. I don't know how failure usually works with beam, so I can't answer much better.

Reply:
Moved the constraints, testing now

Reply:
Testing in https://github.com/obsidiansystems/beam/pull/4

Reply:
OK it passed.